addressing_address_map.go

```diff
@@ -0,0 +1,202 @@
+package cloudflare
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+// AddressMap contains information about an address map.
+type AddressMap struct {
+	ID           string                 `json:"id"`
+	Description  string                 `json:"description"`
+	DefaultSNI   string                 `json:"default_sni"`
+	Enabled      bool                   `json:"enabled"`
+	CanDelete    bool                   `json:"can_delete"`
+	CanModifyIPs bool                   `json:"can_modify_ips"`
+	Memberships  []AddressMapMembership `json:"memberships"`
+	IPs          []AddressMapIP         `json:"ips"`
+	CreatedAt    *time.Time             `json:"created_at"`
```

Comment:
The timestamp fields on all Address Maps endpoints are always non-null, but I am not sure if this is convention in cloudflare-go so I leave it up to you.

Reply:
There are mixed usages of pointers and values for time fields in this repo. I'll make the change if the api does not return null.

Reply:
`time.Time` usage is definitely varied (sorry!). i'm working on documenting type conventions better as people are commonly tripping up on uninitialised values and zero values with this library and i think i can do a better job of explaining where the traps are.

we have [builtin in helpers for type conversion](https://github.com/cloudflare/cloudflare-go/blob/master/convert_types.go) so this should be easy for most to adopt.

Reply:
similarly, i recommend all booleans are boolean pointers so avoid some issues with Terraform.

Reply:
Are you suggesting make all time and boolean fields pointers? And can you please elaborate what would be the issues Terraform run into with non-null fields? 

> `i recommend all booleans are boolean pointers so avoid some issues with Terraform`

Reply:
that would be my suggestion, yes. on the terraform side, there needs to be a way to determine if a field is uninitialised, false or true and the only reliable way of doing that is with pointer fields.

Reply:
i've gone ahead and pushed up some changes to address this